app/locations/controllers.js

```diff
@@ -1,44 +1,19 @@
-const { get, uniqBy } = require('lodash')
+const { get } = require('lodash')
 
 async function locations(req, res, next) {
   const userPermissions = get(req.session, 'user.permissions', [])
 
   let regions = []
-  let userLocations = req.userLocations
-  const supplierId = req.session.user.supplierId
 
   try {
     if (userPermissions.includes('allocation:create')) {
       regions = await req.services.referenceData.getRegions()
     }
-
-    if (supplierId) {
-      userLocations = await req.services.referenceData.getLocationsBySupplierId(
-        supplierId
-      )
-    }
-
-    if (userPermissions.includes('locations:contract_delivery_manager')) {
-      const suppliers = await req.services.referenceData.getSuppliers()
-      const supplierLocations = await Promise.all(
-        suppliers.map(async supplier => {
-          const locations =
-            await req.services.referenceData.getLocationsBySupplierId(
-              supplier.id
-            )
-          return locations
-        })
-      )
-
-      // The locations have been uniqued based on title to prevent
-      // duplicates when multiple suppliers have the same location
-      userLocations = uniqBy(supplierLocations.flat(), 'id')
-    }
   } catch (error) {
     return next(error)
   }
 
-  req.session.user.locations = userLocations
+  const userLocations = req.userLocations
```

Comment:
Having looked at the function that is setting `req.userLocations`, I'm not really sure I'm seeing a good reason for it anymore as its not being used anywhere else.

I wonder if its possible to remove that as part of this change to reduce the amount of abstraction and use something similar as line 4?

```javascript
const userLocations = get(req.session, 'user.locations', [])
```

Reply:
Sounds reasonable! Do you think the canonical way of accessing the user should be `req.session.user` or [`req.user`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/main/common/middleware/set-user.js)?

Reply:
(My preference would be `req.session.user` since it's more explicit.)

Reply:
> Do you think the canonical way of accessing the user should be req.session.user or req.user

That's an excellent question. I think this set of code was introduced before we added `req.user`. The addition of `req.user` was to allow Sentry to automatically add user context to issues and since have started to reference `req.user` instead.

I don't have a strong preference though so happy either way.

Reply:
I've made this change in 1c926ad!